lib/segment/src/types.rs

```diff
@@ -197,6 +198,14 @@ impl PartialEq for ScoredPoint {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize, JsonSchema, Clone)]
+pub struct PointGroup {
+    /// Scored points that have the same group_id
+    pub hits: Vec<ScoredPoint>,
+    /// Key shared by all the hits
+    pub group_id: serde_json::Map<String, Value>,
```

Comment:
`serde_json::Map<String, Value>` is just to produce `{"<key>": "<value>"}` JSON, right? As a minor nitpick, I'd define a `struct GroupId { key: String, value: Value }` with a custom `Serialize` impl that would do that.

Reply:
The reason to use a serde_json::Map was to generate an `object` type on the JsonSchema, as the Value was generating something else. Ideally I'd prefer to skip repeating the group_by field in the result. Do you know a way to achieve this?

Reply:
nvm, already switched to a new GroupId struct